src/main.ts

```diff
@@ -388,95 +433,66 @@ export default class ReadwisePlugin extends Plugin {
   async addBookToRefresh(bookId: string) {
     let booksToRefresh = this.settings.booksToRefresh;
     booksToRefresh.push(bookId);
+    console.log(`Readwise Official plugin: added book id ${bookId} to refresh later`);
     this.settings.booksToRefresh = booksToRefresh;
     await this.saveSettings();
   }
 
-  reimportFile(vault: Vault, fileName: string) {
-    const bookId = this.settings.booksIDsMap[fileName];
+  async removeBooksFromRefresh(bookIds: Array<string> = []) {
+    if (!bookIds.length) return;
+
+    console.log(`Readwise Official plugin: removing book ids ${bookIds.join(', ')} from refresh list`);
+    this.settings.booksToRefresh = this.settings.booksToRefresh.filter(n => !bookIds.includes(n));
+    await this.saveSettings();
+  }
+
+  async reimportFile(vault: Vault, fileName: string) {
     try {
-      fetch(
-        `${baseURL}/api/refresh_book_export`,
-        {
-          headers: {...this.getAuthHeaders(), 'Content-Type': 'application/json'},
-          method: "POST",
-          body: JSON.stringify({exportTarget: 'obsidian', books: [bookId]})
-        }
-      ).then(response => {
-        if (response && response.ok) {
-          let booksToRefresh = this.settings.booksToRefresh;
-          this.settings.booksToRefresh = booksToRefresh.filter(n => ![bookId].includes(n));
-          this.saveSettings();
-          vault.delete(vault.getAbstractFileByPath(fileName));
-          this.startSync();
-        } else {
-          this.notice("Failed to reimport. Please try again", true);
-        }
-      });
+      this.notice("Deleting and reimporting file...", true);
+      await vault.delete(vault.getAbstractFileByPath(fileName));
+      const bookId = this.settings.booksIDsMap[fileName];
+      await this.refreshBookExport([bookId]);
     } catch (e) {
       console.log("Readwise Official plugin: fetch failed in Reimport current file: ", e);
     }
   }
 
-  startSync() {
+  async startSync() {
     if (this.settings.isSyncing) {
       this.notice("Readwise sync already in progress", true);
     } else {
       this.settings.isSyncing = true;
-      this.saveSettings();
-      this.requestArchive();
+      await this.saveSettings();
+      await this.requestArchive();
     }
     console.log("Readwise Official plugin: started sync");
   }
 
   async onload() {
-    // @ts-ignore
+    await this.loadSettings();
+
+    // @ts-expect-error - no type for isMobile
     if (!this.app.isMobile) {
       this.statusBar = new StatusBar(this.addStatusBarItem());
       this.registerInterval(
         window.setInterval(() => this.statusBar.display(), 1000)
       );
     }
-    await this.loadSettings();
-    this.refreshBookExport = debounce(

```

Comment:
Yup. `loadSettings` is moved up to the top of `onLoad` and `debounce` no longer needed because (i think) it was mostly being fired by some excessive calls that were happening elsewhere (the `on(delete)`, maybe?) + more thorough `async`/`await`.